src/i_video.c

```diff
@@ -1029,27 +1029,29 @@ byte I_GetPaletteIndex(byte *palette, int r, int g, int b)
 {
     byte best;
     int best_diff, diff;
-    int i;
+    int i, dr, dg, db;
 
     best = 0;
     best_diff = INT_MAX;
 
     for (i = 0; i < 256; ++i)
     {
```

Comment:
Looks very good! Do you have a benchmark for this?

Reply:
About twice as fast. I was just timing the conversion of a large 32 bit [TITLEPIC.zip](https://github.com/user-attachments/files/16736624/TITLEPIC.zip)

Reply:
Impressive, thank you!